Added movie search engine example (GSoC)

YASH-GU24 opened the pull request.

sebawita commented on movies-search-engine/README.md:
```diff
@@ -0,0 +1,33 @@
+The Dataset used for this example can be found here: https://www.kaggle.com/datasets/yashgupta24/48000-movies-dataset \
+It is data of over 48,000+ movies scraped from IMBD website.
+
+This is a demo example to show how to make a search engine for moveis using weaviate. \
```
moveis typo

sebawita commented on movies-search-engine/README.md:
```diff
@@ -0,0 +1,33 @@
+
+This example uses HTML,CSS,Js for frontend and NodeJs for the backend. 
+
+Follow the following steps to reproduce the example \
+1.Download the dataset from https://www.kaggle.com/datasets/yashgupta24/48000-movies-dataset and paste it in the directory of example \
+2.Make sure the weaviate docker file is running. \
+3.Run the command 'pip install -r requirements.txt' in directory to install all required dependencies \
+4.Now run add_data.py (change path of dataset at line 113 if necessary) to add all the data objects. You can also decrease the number of data objects at line 116 so that it takes less time. \
+5.After adding data run command 'npm install' to install all required node modules.\
+6.After adding data and installing modules run command 'node index.js' and navigate to http://localhost:3000/ to perform searching
```
We need to provide exact commands for those following the steps to execute.

Additionally, there is no `examples` folder. Should that be created at the root of the project?

sebawita commented on movies-search-engine/download_poster.py:
```diff
@@ -0,0 +1,12 @@
+import pandas as pd
+import urllib.request
+data=pd.read_csv("/home/hp/Documents/weaviate-movie-recommendation/final_data.csv")
```
We need to update any references to your local drive, like this:
`/home/hp/Documents/weaviate-movie-recommendation/`
make them relative to the project, (i.e. the `weaviate-movie-recommendation` should be a folder in the project) and provide instructions on where to get the images.

sebawita commented on movies-search-engine/add_data.py:
```diff
@@ -0,0 +1,144 @@
+
+
+#  UNCOMMENT BELOW LINE IF THIS IS NOT THE FIRST TIME YOU ARE ADDING DATA FOR THIS SCHEMA
+# client.schema.delete_class('Movies')
```
Can we make this conditional?
The script could check if the schema for `Movies` exists and if it does then delete it

YASH-GU24 commented:
Hello @sebawita ,
-I fixed the typo error
-By example folder, I meant the folder of this code, basically where add_data.py resides. So i mentioned that in readme also
-The dowload_poster.py had no role in this demo. That was the file that was created for the poster image reverse searching feature which we initially discussed but then dropped. Somehow that file made it in the initial commit but now I have deleted it.
-I made the deleting of the schema class as conditional. It will search for all the schema classes present and delete the 'Movies' class if found.

sebawita commented on movies-search-engine/.gitignore:
```diff
@@ -1,2 +1,3 @@
 node_modules
 package-lock.json
+fina_data.csv
```
shouldn't this be final_data.csv ?

YASH-GU24 replied:
Yeah of course, Sorry for the typo
